crucible-llvm/src/Lang/Crucible/LLVM/Eval.hs

```diff
@@ -30,18 +32,20 @@ assertSideCondition ::
   LLVMSideCondition (RegValue' sym) ->
   IO ()
 assertSideCondition sym (LLVMSideCondition (RV p) ub) =
-  do p' <- annotateUB sym ub p
+  do p' <- annotateUB sym noCallStack ub p
```

Comment:
In order to get a callstack from the LLVM memory here, it looks like

- Expression extension evaluators will need (read-only) access to the `CrucibleState`. Not sure if this is acceptable (e.g., is there an `IORef` in there that would be problematic if it were modified by evaluation of an expression?).
- I'll need to find some way to feed in the `mvar` (memory global variable) into `llvmExtensionEval`. For statements, this is embedded in their syntax. Not sure if the same thing is appropriate or possible here.

Reply:
Extensions already essentially have unlimited access to mutate the crucible state in the statement evaluator, I don't see a problem threading it in here.

I'm guessing you'll have access to the `mvar` as necessary when generating side conditions, so that's probably the way to go, yeah.